app/admin.rb

```diff
@@ -3,24 +3,37 @@
 module Admin
   extend Discordrb::Commands::CommandContainer
 
-  command :adddeath, help_available: false do |event, *parse_string|
+  def self.find_role(event, role)
     role = nil
-    parse_string = parse_string.join(' ')
-    parse_string = parse_string.split('-', 2)
-    name = parse_string[0].chomp
-    reason = parse_string[1].chomp.gsub!(/^\s+/, '')
-
-    puts name
-    puts reason
-
     event.server.roles.each do |server_role|
       next unless server_role.name == 'botadmin'
       role = server_role
-      break
     end
+    return role
+  end
 
+  command :updateprofile, help_available: false do |event|
+    role = find_role(event, 'botadmin')
     break unless event.user.role?(role)
```

Comment:
Actually, forget the previous comment. discordrb already has an efficient way of managing command permissions: https://github.com/meew0/discordrb/wiki/Commands#permissions and http://www.rubydoc.info/gems/discordrb/Discordrb%2FCommands%2FCommandContainer%3Acommand . So it would be best to specify a needed permission level for the command, and give that level to botadmin role.